election/example/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	election "k8s.io/contrib/election/lib"
+
+	"github.com/golang/glog"
+	flag "github.com/spf13/pflag"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	kubectl_util "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+var (
+	flags = flag.NewFlagSet(
+		`elector --election=<name>`,
+		flag.ExitOnError)
+	name      = flags.String("election", "", "The name of the election")
+	id        = flags.String("id", "", "The id of this participant")
+	namespace = flags.String("election-namespace", "default", "The Kubernetes namespace for this election")
+	ttl       = flags.Duration("ttl", 10*time.Second, "The TTL for this election")
+	inCluster = flags.Bool("use-cluster-credentials", false, "Should this request use cluster credentials?")
+	addr      = flags.String("http", "", "If non-empty, stand up a simple webserver that reports the leader state")
+
+	leader = &LeaderData{}
+)
+
+func makeClient() (*client.Client, error) {
+	var cfg *client.Config
+	var err error
+
+	if *inCluster {
+		if cfg, err = client.InClusterConfig(); err != nil {
+			return nil, err
+		}
+	} else {
+		clientConfig := kubectl_util.DefaultClientConfig(flags)
+		if cfg, err = clientConfig.ClientConfig(); err != nil {
+			return nil, err
+		}
+	}
+	return client.New(cfg)
+}
+
+type LeaderData struct {
+	Name string `json:"name"`
+}
+
+func webHandler(res http.ResponseWriter, req *http.Request) {
+	data, err := json.Marshal(leader)
+	if err != nil {
+		res.WriteHeader(http.StatusInternalServerError)
+		res.Write([]byte(err.Error()))
+		return
+	}
+	res.WriteHeader(http.StatusOK)
+	res.Write(data)
+}
+
+func validateFlags() {
+	if len(*id) == 0 {
+		glog.Fatal("--id cannot be empty")
+	}
+	if len(*name) == 0 {
+		glog.Fatal("--election cannot be empty")
+	}
+}
+
+func main() {
+	flags.Parse(os.Args)
+	validateFlags()
+
+	kubeClient, err := makeClient()
+	if err != nil {
+		glog.Fatalf("error connecting to the client: %v", err)
+	}
+
+	fn := func(str string) {
+		leader.Name = str
+		fmt.Printf("%s is the leader\n", leader.Name)
+	}
+
+	e, err := election.NewElection(*name, *id, *namespace, *ttl, fn, kubeClient)
+	if err != nil {
+		glog.Fatalf("failed to create election: %v", err)
+	}
+	go e.Run()
```

Comment:
the e.Run() should be warpped in an Until Forever. A single election client does one cycle of candiate->aquire master->lose master and is expected to compelete.